src/EVM/Expr.hs

```diff
@@ -1567,38 +1570,54 @@ data ConstState = ConstState
   }
   deriving (Show)
 
--- | Checks if a conjunction of propositions is definitely unsatisfiable
-isUnsat :: [Prop] -> Bool
-isUnsat ps = Prelude.not $ oneRun ps (ConstState mempty True)
+-- | Performs constant propagation
```

Comment:
It's cool that you managed to generalize this function!

Reply:
Thanks! I was pretty happy about myself actually. Kinda weird it worked so easily.